tests/test_elements/test_mol2mol/test_mol2mol_filter.py

```diff
@@ -14,68 +23,246 @@
 SMILES_CL_BR = "NC(Cl)(Br)C(=O)O"
 SMILES_METAL_AU = "OC[C@H]1OC(S[Au])[C@H](O)[C@@H](O)[C@@H]1O"
 
+SMILES_LIST = [
+    SMILES_ANTIMONY,
+    SMILES_BENZENE,
+    SMILES_CHLOROBENZENE,
+    SMILES_METAL_AU,
+    SMILES_CL_BR,
+]
+
 
 class MolFilterTest(unittest.TestCase):
     """Unittest for MolFilter, which invalidate molecules based on criteria defined in the respective filter."""
 
     def test_element_filter(self) -> None:
-        """Test if molecules are filtered correctly by allowed chemical elements.
-
-        Returns
-        -------
-        None
-        """
-        smiles2mol = SmilesToMol()
-        default_atoms = {
-            1,
-            5,
-            6,
-            7,
-            8,
-            9,
-            14,
-            15,
-            16,
-            17,
-            34,
-            35,
-            53,
+        """Test if molecules are filtered correctly by allowed chemical elements."""
+        default_atoms_dict = {
+            1: (0, None),
+            5: (0, None),
+            6: (0, None),
+            7: (0, None),
+            8: (0, None),
+            9: (0, None),
+            14: (0, None),
+            15: (0, None),
+            16: (0, None),
+            17: (0, None),
+            34: (0, None),
+            35: (0, None),
+            53: (0, None),
         }
 
         element_filter = ElementFilter()
-        self.assertEqual(element_filter.allowed_element_numbers, default_atoms)
-        mol2smiles = MolToSmiles()
-        error_filter = ErrorFilter.from_element_list(
-            [smiles2mol, element_filter, mol2smiles]
-        )
+        self.assertEqual(element_filter.allowed_element_numbers, default_atoms_dict)
         pipeline = Pipeline(
             [
-                ("Smiles2Mol", smiles2mol),
+                ("Smiles2Mol", SmilesToMol()),
                 ("ElementFilter", element_filter),
-                ("Mol2Smiles", mol2smiles),
-                ("ErrorFilter", error_filter),
+                ("Mol2Smiles", MolToSmiles()),
+                ("ErrorFilter", ErrorFilter()),
             ],
         )
-        filtered_smiles = pipeline.fit_transform(
+        filtered_smiles = pipeline.fit_transform(SMILES_LIST)
+        self.assertEqual(
+            filtered_smiles, [SMILES_BENZENE, SMILES_CHLOROBENZENE, SMILES_CL_BR]
+        )
+        pipeline.set_params(
+            ElementFilter__allowed_element_numbers={6: 6, 1: (5, 6), 17: (0, 1)}
+        )
+        filtered_smiles_2 = pipeline.fit_transform(SMILES_LIST)
+        self.assertEqual(filtered_smiles_2, [SMILES_BENZENE, SMILES_CHLOROBENZENE])
+
+    def test_complex_filter(self) -> None:
```

Comment:
we also need a test_get_set_params_complex_filter.

I would advocate for implementing one TestClass for each FilterClass.  @JochenSiegWork what do you think?

Reply:
The critical part is if we can write it as a json file and load the the object from the file.  
A nice to have would be if the parameters of the subfilters can be accessed via the get/set_params from the filter, à la `filter.set_params(subfiltername__allowed_element_numbers=[...])`

Reply:
How does the write as json file work (where is the code reference)? I do not see the direct relation of get/set_params and save as json (also, see the link in my previous comment, probably this is the same issue)

Reply:
https://github.com/basf/MolPipeline/blob/main/molpipeline/utils/json_operations.py#L218

https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/base.py#L221

Reply:
Believe I implemented this (quite hacky, see additional command), tests look good, feel free to check